### PR TITLE
Generate function name, and description based on input and return values

### PR DESCRIPTION
**Pull request by bokimilinkovic**

### CODE REVIEW

File: test-crawler/collector/collector.go [hunk 1]
```diff
@@ -126,3 +129,29 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
 
 	return "", nil, nil
 }
+
+// RegenerateName regenerates name adding underscore and usecase index
+// ex: name: ListenProofRequest -> LISTEN_PROOF_REQUEST_001
+func (f *FunctionAnnotation) RegenerateName() {
```
Comment (brdji) [inline test-crawler/collector/collector.go:135]: I would rather have this function return the value of the new name, so the caller can set it.
We are mutating the structure right now (ie. overwriting the value of `f.Name`), but in the future, we will create a new structure, to keep the logic separate.

File: test-crawler/collector/collector.go [hunk 2]
```diff
@@ -126,3 +129,29 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
...
+			buf.WriteRune('_')
+		}
+		buf.WriteRune(rune)
+	}
+
+	f.Name = fmt.Sprintf("%s_%s", strings.ToUpper(buf.String()), usecase)
+}
+
+// GenerateDescription generates description based on inputParams, returnParams
+func (f *FunctionAnnotation) RegenerateDescription() {
```
Comment (brdji) [inline test-crawler/collector/collector.go:149]: See above: return the new value of the description.
Also, we should return a default value when input params / return values are not set. Something like "Function description not set."

File: test-crawler/collector/collector.go [hunk 3]
```diff
@@ -126,3 +129,31 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
 
 	return "", nil, nil
 }
+
+// GeneratMethodName generates name adding underscore and usecase index
+// ex: name: ListenProofRequest -> LISTEN_PROOF_REQUEST_001
+func GeneratMethodName(funcName string) string {
```
Comment (brdji) [inline test-crawler/collector/collector.go:135]: Typo: `Generate` instead of `Generate`

File: test-crawler/collector/collector.go [hunk 4]
```diff
@@ -126,3 +129,31 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
...
+}
+
+// GenerateMethodDescription generates description based on inputParams, returnParams
+func GenerateMethodDescription(f FunctionAnnotation) string {
+	var dsc string
+	if f.Description == "" {
+		dsc = "Function description not set."
+		if (f.InputParams != "" && f.InputParams != "()") && f.ReturnValues != "" {
+			dsc = fmt.Sprintf(`Given a %s, returns %s`, f.InputParams, f.ReturnValues)
+		}
```
Comment (brdji) [inline test-crawler/collector/collector.go:155]: Why did you remove the case when the function has no input values, but has a return value?

Comment (bokimilinkovic) [reply test-crawler/collector/collector.go:155]: I think that's dummy description, better to be default value than just simple 'returns ...' , since we will have 3 cases:
1. GoDoc description if exists
2. Given a input params, returns ...
3. Default value

**Review by brdji — APPROVED** [review]
